meow/eme/propagate.py

```diff
@@ -103,18 +102,17 @@ def compute_mode_amplitudes(u, v, m, excitation_l, excitation_r):
     return forward, backward
 
 
-def plot_fields(modes, forwards, backwards, y, z, lim=1):
+def plot_fields(modes, cells, forwards, backwards, y, z, lim=1):
```

Comment:
Sidenote independent of PR. Should make this compatible to multiple meshes along z?

Reply:
not sure what you mean by that?

Reply:
I get the mesh from the first cell and assume that the mesh is the same for all cells. However, I have seen that the option exists to have a different mesh on every Cell. (How do we calculate the overlap in these cases btw?)

Reply:
Yeah, I left the option open for different meshes in each cell, but it's clearly not fully supported. I think to calculate the overlap you probably have to interpolate the modes from the smaller mesh to fit on the bigger mesh or something like that. In any case... not really a priority right now.

That said, maybe a simple check to ensure all the meshes are the same would be good?

Reply:
Jep, seems like a good check to avoid unwanted behavior. I think we should also check in the overlap calculations